daemon/emer-daemon.c

```diff
@@ -141,7 +142,8 @@ typedef struct _EmerDaemonPrivate
 {
   guint network_send_interval;
   GQueue *upload_queue;
-  gboolean uploading;
+  /* Non-NULL iff an upload is in flight */
+  GCancellable *current_upload_cancellable;
```

Comment:
Who owns this cancellable? Looks like you unref it when freeing the network callback data below